### PR TITLE
Use the injector to instantiate a new Page.

### PR DESCRIPTION
**Pull request by Firesphere**
This way, a new project can have a namespaced Page Class.
It can then be mapped in the YML config with

```
SilverStripe\Core\Injector\Injector:
  Page:
    class: My\Namespace\Page
```

### CODE REVIEW

File: code/Model/SiteTree.php [hunk 1]
```diff
@@ -1515,7 +1516,7 @@ public function requireDefaultRecords() {
 		// default pages
 		if(static::class == self::class && $this->config()->create_default_pages) {
 			if(!SiteTree::get_by_link(RootURLController::config()->default_homepage_link)) {
-				$homepage = new Page();
+				$homepage = Injector::inst()->get('Page', false);
```
Comment (dhensby) [inline code/Model/SiteTree.php]: I think using `create` rather than `get` is a bit cleaner here

Comment (Firesphere) [inline code/Model/SiteTree.php]: Ah, right, semantically better indeed.

**Comment by Firesphere** [comment]
Updating as per @dhensby's comment, and also applied the same logic to the ErrorPage, to keep things consistent.

**Comment by Firesphere** [comment]
Breaking of build is unrelated to this PR.

File: code/Model/ErrorPage.php [hunk 2]
```diff
@@ -143,7 +143,7 @@ protected function requireDefaultRecordFixture($defaultData) {
 		$page = ErrorPage::get()->filter('ErrorCode', $code)->first();
 		$pageExists = !empty($page);
 		if(!$pageExists) {
-			$page = new ErrorPage($defaultData);
+			$page = Injector::inst()->create('SilverStripe\\CMS\\Model\\ErrorPage', $defaultData);
```
Comment (kinglozzer) [inline code/Model/ErrorPage.php]: `ErrorPage::class` :)

Comment (Firesphere) [inline code/Model/ErrorPage.php]: Does that include the namespace? I did not know that!
Thanks, will update :)

**Comment by chillu** [comment]
Given the accepted RFC option is to use a small helper file which defines a namespaced `Page` class (https://github.com/silverstripe/silverstripe-framework/issues/5844), I don't see the need for this PR any more. There's going to be many other use cases where `new Page()` is used directly, and mandating `Injector` because of a relatively obscure API quirk for this particular instanciation feels a bit dirty. `Injector` should only be used for "services" in the wider sense - but not for every core object (e.g. `Page`).

**Pull request closed without merging**